### PR TITLE
py-docutils: improvement

### PR DESCRIPTION
**Pull request by citibeth**
Adds dependencies that improve the usefulness of Docutils.

### CODE REVIEW

File: var/spack/repos/builtin/packages/doxygen/package.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ class Doxygen(Package):
     depends_on("graphviz", when="+graphviz", type='run')
 
     def install(self, spec, prefix):
-        cmake('.', *std_cmake_args)
```
Comment (davydden) [inline var/spack/repos/builtin/packages/doxygen/package.py]: are these changes necessary? I think it has been fix a while ago. If so, please remove all 
`cmake() -> which('cmake')()` from this PR so that it's easier to review.

File: var/spack/repos/builtin/packages/mumps/package.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ class Mumps(Package):
     homepage = "http://mumps.enseeiht.fr"
     url      = "http://mumps.enseeiht.fr/MUMPS_5.0.1.tar.gz"
 
-    version('5.0.1', 'b477573fdcc87babe861f62316833db0')
+    version('5.0.1', 'b477573fdcc87babe861f62316833db0', url='http://pkgs.fedoraproject.org/repo/pkgs/MUMPS/MUMPS_5.0.1.tar.gz/md5/b477573fdcc87babe861f62316833db0/MUMPS_5.0.1.tar.gz')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/mumps/package.py]: official 'url` works just fine. No need for a fedora mirror at the moment.

File: var/spack/repos/builtin/packages/petsc/package.py [hunk 3]
```diff
@@ -127,7 +134,9 @@ def install(self, spec, prefix):
                 'complex' if '+complex' in spec else 'real'),
             '--with-shared-libraries=%s' % ('1' if '+shared' in spec else '0'),
             '--with-debugging=%s' % ('1' if '+debug' in spec else '0'),
-            '--with-blas-lapack-dir=%s' % spec['lapack'].prefix
+            '--with-lapack-lib=%s' % spec['lapack'].lapack_static_lib,
```
Comment (davydden) [inline var/spack/repos/builtin/packages/petsc/package.py]: i think we already discussing this in https://github.com/LLNL/spack/pull/1538#discussion_r75065373 . Why creating **two PR's with the same changes** which also do not reflect the title of this PR? Please keep things separately. Since you already modify `petsc` similarly in another PR, do not change the file in this one.

File: var/spack/repos/builtin/packages/py-scipy/package.py [hunk 4]
```diff
@@ -45,7 +45,18 @@ def install(self, spec, prefix):
             env['ATLAS'] = join_path(
                 spec['atlas'].prefix.lib, 'libatlas.' + dso_suffix)
         else:
-            env['BLAS']   = spec['blas'].blas_shared_lib
-            env['LAPACK'] = spec['lapack'].lapack_shared_lib
+            blas_spec = spec['blas']
```
Comment (davydden) [inline var/spack/repos/builtin/packages/py-scipy/package.py]: same here. I saw this in another PR of yours.

File: var/spack/repos/builtin/packages/suite-sparse/package.py [hunk 5]
```diff
@@ -23,7 +23,7 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 ##############################################################################
 from spack import *
-
+import os
```
Comment (davydden) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: you don't use 'os', do you?

**Comment by davydden** [comment]
i would strongly suggest splitting this huge PR with 50 files changed into several. You already have some of the changes elsewhere in onepened PR's. I am not a Spack developer, but my general suggestion would be: keep things simple, on topic and related within a single PR. For the current changes i just fail to see the relationship between them. And there is certainly no need to duplicate changes within several PRs.

**Comment by davydden** [comment]
it looks like you branch was not rebased on `develop`, there are commits from May by yourself and also @tgamblin , which is likely not your intention here.

File: var/spack/repos/builtin/packages/doxygen/package.py (hunk 1 above)
Comment (davydden) [inline var/spack/repos/builtin/packages/doxygen/package.py]: i think you know the answer already https://groups.google.com/forum/?fromgroups#!topic/spack/gxizu9g40oU :smile:

**Comment by citibeth** [comment]
Sorry, fat fingers on the git for what was supposed to be a 2-line change.  Will do it again later.
